assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js

```diff
@@ -167,7 +169,20 @@ function PaginatedMediaGallery({
       </div>
     );
 
-  const displayLoadingPill = isMediaLoading && hasMore;
+  const [showLoadingPill, setShowLoadingPill] = useState(false);
+
+  useEffect(() => {
+    if (isMediaLoading) {
+      const showLoadingTimeout = setTimeout(() => {
+        setShowLoadingPill(isMediaLoading);
+      }, SHOW_LOADING_PILL_DELAY_MS);
+      return () => clearTimeout(showLoadingTimeout);
+    }
+    setShowLoadingPill(false);
+    return undefined;
```

Comment:
Is this necessary?

Reply:
Unfortunately it is. If you return sth from an effect (to be executed on unmount), you have yto be consistent and always return sth, undefined in the case that you want nothing to occur in that situation.

Reply:
Ah, is `return undefined` preferred over `return () => {}`?

Reply:
Yes, the linter explicitly said to return undefined in that case.